src/Microsoft.AspNetCore.Sockets.Client.Http/HttpConnection.cs

```diff
@@ -116,10 +117,12 @@ private Task StartAsyncCore()
                     if (t.IsFaulted)
                     {
                         _startTcs.SetException(t.Exception.InnerException);
+                        _closedTcs.TrySetException(t.Exception.InnerException);
```

Comment:
do we have a test for this condition and (possibly) the next condition (not sure if `t.IsCanceled` is possible to hit)?